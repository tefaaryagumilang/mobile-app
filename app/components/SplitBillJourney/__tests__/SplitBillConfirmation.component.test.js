import React from 'react';
import renderer from 'react-test-renderer';
import SplitBillConfirmation from '../SplitBillConfirmation.component';
import {Provider} from 'react-redux';
import {createStore} from 'redux';
import {reduxForm} from 'redux-form';

const store = createStore(() => ({}));
const DecoratedForm = reduxForm({form: 'testForm'})(SplitBillConfirmation);

describe('SplitBillConfirmation: form page', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Provider store={store}><DecoratedForm/></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
