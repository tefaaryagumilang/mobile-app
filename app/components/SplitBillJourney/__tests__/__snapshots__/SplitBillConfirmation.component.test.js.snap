// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitBillConfirmation: form page renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <KeyboardAwareScrollView
    contentContainerStyle={
      Array [
        Object {
          "flexGrow": 1,
          "justifyContent": "space-between",
          "paddingBottom": 20,
        },
      ]
    }
    enableOnAndroid={true}
    extraScrollHeight={100}
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "backgroundColor": "#F1F3F6",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FA3F70",
          "borderColor": "transparent",
          "borderWidth": 1,
          "height": 133.4,
          "marginBottom": 533.6,
          "position": "absolute",
          "width": 750,
        }
      }
    />
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 15,
            "marginHorizontal": 37.5,
          }
        }
      >
        <View
          style={
            Object {
              "paddingHorizontal": 20,
              "paddingVertical": 20,
            }
          }
        >
          <View>
            <View>
              <Image
                source="icon_rp.png"
                style={
                  Object {
                    "height": 30,
                    "left": 25,
                    "position": "absolute",
                    "top": 25,
                    "width": 30,
                  }
                }
              />
              <View>
                <View>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                          "paddingHorizontal": 20,
                        },
                        Object {
                          "borderColor": "#ccc",
                          "borderRadius": 10,
                          "borderWidth": 1,
                          "marginVertical": 5,
                        },
                      ]
                    }
                  >
                    <View>
                      <View>
                        <Text
                          style={
                            Object {
                              "color": "#7D7F9E",
                              "fontFamily": "roboto",
                              "marginBottom": -10,
                              "paddingHorizontal": 50,
                              "paddingTop": 10,
                            }
                          }
                        >
                          Nominal
                        </Text>
                      </View>
                      <TextInput
                        allowFontScaling={true}
                        autoCapitalize="none"
                        disabled={false}
                        disabledLineType="solid"
                        editable={true}
                        keyboardType="numeric"
                        label=""
                        maxLength={10}
                        onBlur={[Function]}
                        onChangeText={[Function]}
                        onFocus={[Function]}
                        placeholder="0"
                        rejectResponderTermination={true}
                        style={
                          Object {
                            "color": "#000000",
                            "fontFamily": "roboto",
                            "fontSize": 18,
                            "fontWeight": "700",
                            "height": 50,
                            "marginBottom": -5,
                            "paddingHorizontal": 50,
                            "width": 630,
                          }
                        }
                        theme="primary"
                        tintColor="#D72F33"
                        underlineColorAndroid="transparent"
                        value=""
                      />
                    </View>
                    <View>
                      <Text
                        allowFontScaling={false}
                        style={
                          Array [
                            Object {
                              "color": undefined,
                              "fontSize": 18,
                            },
                            Object {
                              "color": "#000000",
                            },
                            Object {
                              "fontFamily": "simas",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                      >
                        
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View>
            <View>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "paddingHorizontal": 20,
                    },
                    Object {
                      "borderColor": "#ccc",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "marginVertical": 5,
                    },
                  ]
                }
              >
                <View>
                  <View>
                    <Text
                      numberOfLines={1}
                      style={
                        Object {
                          "color": "#ccc",
                          "fontFamily": "roboto",
                          "fontSize": 14,
                          "left": 0,
                          "position": "absolute",
                          "top": 15,
                        }
                      }
                    >
                      Deskripsi
                    </Text>
                  </View>
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    disabled={false}
                    disabledLineType="solid"
                    editable={true}
                    label=""
                    maxLength={25}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder=""
                    rejectResponderTermination={true}
                    style={
                      Object {
                        "color": "#2F476D",
                        "fontFamily": "roboto",
                        "fontSize": 14,
                        "fontWeight": "700",
                        "height": 50,
                        "marginBottom": -5,
                        "marginTop": 10,
                        "paddingRight": 25,
                        "width": 630,
                      }
                    }
                    theme="primary"
                    tintColor="#D72F33"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
                <View>
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 18,
                        },
                        Object {
                          "color": "#000000",
                        },
                        Object {
                          "fontFamily": "simas",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View>
              <View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "paddingHorizontal": 20,
                      },
                      Object {
                        "borderColor": "#ccc",
                        "borderRadius": 10,
                        "borderWidth": 1,
                        "marginVertical": 5,
                      },
                    ]
                  }
                >
                  <View>
                    <View>
                      <Text
                        numberOfLines={1}
                        style={
                          Object {
                            "color": "#ccc",
                            "fontFamily": "roboto",
                            "fontSize": 14,
                            "left": 0,
                            "position": "absolute",
                            "top": 15,
                          }
                        }
                      >
                        Lampiran (jika ada)
                      </Text>
                    </View>
                    <TextInput
                      allowFontScaling={true}
                      autoCapitalize="none"
                      baseColor="#949494"
                      disabled={true}
                      disabledLineType="solid"
                      editable={false}
                      label=""
                      onBlur={[Function]}
                      onChangeText={[Function]}
                      onFocus={[Function]}
                      placeholder=""
                      rejectResponderTermination={true}
                      style={
                        Object {
                          "color": "#2F476D",
                          "fontFamily": "roboto",
                          "fontSize": 14,
                          "fontWeight": "700",
                          "height": 50,
                          "marginBottom": -5,
                          "marginTop": 10,
                          "paddingRight": 25,
                          "width": 630,
                        }
                      }
                      textColor="#949494"
                      theme="primary"
                      tintColor="#949494"
                      underlineColorAndroid="transparent"
                      value=""
                    />
                  </View>
                  <View>
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": undefined,
                            "fontSize": 18,
                          },
                          Object {
                            "color": "#000000",
                          },
                          Object {
                            "fontFamily": "simas",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "position": "absolute",
                  "right": 10,
                  "top": 25,
                  "width": 25,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 20,
                    },
                    Object {
                      "color": "#2F476D",
                    },
                    Object {
                      "fontFamily": "simas",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <View>
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#ccc",
                    "justifyContent": "center",
                    "marginBottom": 5,
                    "paddingHorizontal": 20,
                  },
                  Object {},
                  Object {
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "color": "#ccc",
                    "marginVertical": 5,
                  },
                ]
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#ccc",
                      "fontSize": 12,
                      "marginBottom": -15,
                      "paddingVertical": 5,
                    }
                  }
                >
                  Rekening penerimaan
                </Text>
              </View>
              <View>
                <View
                  accessible={true}
                  clickable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 50,
                      "justifyContent": "space-between",
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Array [
                          Object {
                            "flex": 1,
                          },
                          Object {
                            "color": "#000",
                            "fontSize": 14,
                          },
                          Object {
                            "fontWeight": "700",
                          },
                          Object {
                            "color": "#2F476D",
                          },
                        ],
                        Object {},
                      ]
                    }
                  >
                    Rekening penerimaan
                  </Text>
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        Array [
                          Object {
                            "color": "#2F476D",
                            "paddingRight": 15,
                          },
                          Array [
                            Object {
                              "alignSelf": "flex-end",
                              "bottom": 10,
                              "color": "#2F476D",
                              "left": 5,
                              "transform": Array [
                                Object {
                                  "rotate": "90deg",
                                },
                              ],
                            },
                            Object {},
                          ],
                        ],
                        Object {
                          "fontFamily": "simas",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Modal
                  animationType="none"
                  hardwareAccelerated={false}
                  hideOnBack={false}
                  onBackButtonPress={[Function]}
                  onRequestClose={[Function]}
                  transparent={true}
                  visible={false}
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 0,
                        "height": 1334,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "width": 750,
                      }
                    }
                  />
                  <View
                    hideOnBack={false}
                    onBackButtonPress={[Function]}
                    style={
                      Object {
                        "flex": 1,
                        "justifyContent": "center",
                        "margin": 0,
                        "padding": 0,
                      }
                    }
                    transparent={true}
                  >
                    <View
                      contentContainerStyle={
                        Object {
                          "flex": 1,
                        }
                      }
                      extraHeight={120}
                      keyboardShouldPersistTaps="handled"
                      scrollEnabled={false}
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        accessible={true}
                        clickable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "rgba(37, 8, 10, 0.78)",
                              "flex": 1,
                              "justifyContent": "center",
                              "padding": 20,
                            },
                            Object {
                              "justifyContent": "flex-end",
                              "padding": 0,
                            },
                          ]
                        }
                      >
                        <View
                          accessible={true}
                          clickable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#FFFFFF",
                                "borderRadius": 20,
                                "padding": 20,
                              },
                              Object {
                                "backgroundColor": "transparent",
                                "padding": 0,
                              },
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#FFFFFF",
                                "borderRadius": 20,
                                "padding": 20,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#ededed",
                                "justifyContent": "center",
                                "padding": 0,
                                "width": 750,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                  "height": 50,
                                  "justifyContent": "space-between",
                                  "paddingHorizontal": 20,
                                  "width": 750,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "flexDirection": "row",
                                  }
                                }
                              >
                                <View
                                  accessible={true}
                                  clickable={true}
                                  isTVSelectable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "opacity": 1,
                                      "paddingRight": 20,
                                    }
                                  }
                                >
                                  <Text
                                    allowFontScaling={false}
                                    style={
                                      Array [
                                        Object {
                                          "color": undefined,
                                          "fontSize": 20,
                                        },
                                        Object {
                                          "color": "#007AFF",
                                          "transform": Array [
                                            Object {
                                              "rotate": "180deg",
                                            },
                                          ],
                                        },
                                        Object {
                                          "fontFamily": "simas",
                                          "fontStyle": "normal",
                                          "fontWeight": "normal",
                                        },
                                        Object {},
                                      ]
                                    }
                                  >
                                    
                                  </Text>
                                </View>
                                <View
                                  accessible={true}
                                  clickable={true}
                                  isTVSelectable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "opacity": 1,
                                      "paddingLeft": 20,
                                    }
                                  }
                                >
                                  <Text
                                    allowFontScaling={false}
                                    style={
                                      Array [
                                        Object {
                                          "color": undefined,
                                          "fontSize": 20,
                                        },
                                        Object {
                                          "color": "#007AFF",
                                        },
                                        Object {
                                          "fontFamily": "simas",
                                          "fontStyle": "normal",
                                          "fontWeight": "normal",
                                        },
                                        Object {},
                                      ]
                                    }
                                  >
                                    
                                  </Text>
                                </View>
                              </View>
                              <View
                                accessible={true}
                                clickable={true}
                                isTVSelectable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "opacity": 1,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#007AFF",
                                      "fontSize": 16,
                                      "fontWeight": "700",
                                    }
                                  }
                                >
                                  Selesai
                                </Text>
                              </View>
                            </View>
                            <View>
                              <View
                                style={
                                  Object {
                                    "width": 750,
                                  }
                                }
                              >
                                <RCTPicker
                                  items={
                                    Array [
                                      Object {
                                        "label": "Rekening penerimaan",
                                        "textColor": undefined,
                                        "value": "Rekening penerimaan",
                                      },
                                    ]
                                  }
                                  onChange={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  selectedIndex={0}
                                  style={
                                    Array [
                                      Object {
                                        "height": 216,
                                      },
                                      Object {
                                        "backgroundColor": "#FFFFFF",
                                      },
                                    ]
                                  }
                                />
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </Modal>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "marginTop": 10,
          "paddingBottom": 20,
          "paddingHorizontal": 40,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#2F476D",
            "fontFamily": "roboto",
            "fontSize": 16,
            "fontWeight": "700",
          }
        }
      >
        Anggota
      </Text>
      <Text
        style={
          Object {
            "color": "#7A7D9E",
            "fontFamily": "roboto",
            "fontSize": 14,
            "fontWeight": "300",
            "paddingLeft": 5,
          }
        }
      >
        (0
         
        orang)
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 20,
          "marginBottom": 20,
          "marginHorizontal": 20,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "paddingVertical": 10,
              }
            }
          >
            Metode pembagian
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingVertical": 10,
              }
            }
          >
            <View>
              <View
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "#FA3F70",
                    "borderColor": "#FFFFFF",
                    "borderRadius": 20,
                    "borderWidth": 1,
                    "opacity": 1,
                    "paddingHorizontal": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#FFFFFF",
                    }
                  }
                >
                  Rata-rata
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "paddingLeft": 10,
                }
              }
            >
              <View
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#ccc",
                    "borderRadius": 20,
                    "borderWidth": 1,
                    "opacity": 1,
                    "paddingHorizontal": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#ccc",
                    }
                  }
                >
                  Tulis Nominal
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 20,
            }
          }
        >
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "paddingHorizontal": 10,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 20,
                    },
                    Object {
                      "color": "#2F476D",
                    },
                    Object {
                      "fontFamily": "simas",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <View
                style={
                  Object {
                    "flex": 0,
                    "marginLeft": 10,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontWeight": "700",
                      },
                      Object {
                        "color": "#2F476D",
                        "fontFamily": "roboto",
                      },
                    ]
                  }
                >
                  Tambah anggota baru
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </KeyboardAwareScrollView>
  <View
    style={
      Object {
        "backgroundColor": "#F1F3F6",
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FA3F70",
              "borderRadius": 30,
              "flexDirection": "row",
              "height": 47,
              "justifyContent": "center",
              "paddingHorizontal": 15,
            },
            Object {
              "marginVertical": 10,
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "#FF8E8E",
              "borderRadius": 30,
              "flexDirection": "row",
              "height": 47,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          SPLIT BILL
        </Text>
      </View>
    </View>
  </View>
</View>
`;
